src/autopas/tuning/tuningStrategy/decisionTreeTuning/DecisionTreeTuning.cpp

```diff
@@ -0,0 +1,168 @@
+/**
+ * @file DecisionTreeTuning.cpp
+ * @author Abdulkadir Pazar
+ * @date 20.06.24
+ */
+
+#include "DecisionTreeTuning.h"
+
+#include <Python.h>
+
+#include <iostream>
+#include <sstream>
+
+namespace autopas {
+
+// Constructor
+DecisionTreeTuning::DecisionTreeTuning(const std::set<Configuration> &searchSpace, const std::string &modelFileName)
+    : _configurations(searchSpace), _modelFileName(modelFileName), _pFunc(nullptr) {
+  // Initialize Python Interpreter
+  Py_Initialize();
+  loadScript();
+}
+
+// Destructor
+DecisionTreeTuning::~DecisionTreeTuning() {
+  // Finalize Python Interpreter
+  if (_pFunc) {
+    Py_DECREF(_pFunc);  // Clean up the reference to _pFunc
+  }
+  Py_Finalize();
+}
+
+void DecisionTreeTuning::loadScript() {
+  // Add the current directory to the Python path
+  PyObject *sysPath = PySys_GetObject((char *)"path");
+  PyObject *cwd = PyUnicode_FromString(".");
+  PyList_Append(sysPath, cwd);
+  Py_DECREF(cwd);
+
+  // Load the fixed Python script predict.py
+  PyObject *pName = PyUnicode_DecodeFSDefault("predict");
+  if (!pName) {
+    throw std::runtime_error("Failed to convert script name to Python string.");
+  }
+
+  PyObject *pModule = PyImport_Import(pName);
+  Py_DECREF(pName);
+
+  if (!pModule) {
+    PyErr_Print();
+    throw std::runtime_error("Failed to load Python module: predict.py");
+  }
+
+  PyObject *pDict = PyModule_GetDict(pModule);
+  Py_DECREF(pModule);
+
+  if (!pDict) {
+    PyErr_Print();
+    throw std::runtime_error("Failed to get dictionary from Python module: predict.py");
+  }
+
+  PyObject *pFunc = PyDict_GetItemString(pDict, "main");
+  if (!pFunc || !PyCallable_Check(pFunc)) {
+    PyErr_Print();
+    throw std::runtime_error("Failed to find function 'main' in module: predict.py");
+  }
+
+  _pFunc = pFunc;
+  Py_INCREF(_pFunc);
```

Comment:
Briefly describe what this does or provide a reference. Just because I'd expect most AutoPas developers to be unfamiliar with Python/C Interfacing